### DISCUSSION TITLE
Introduction to LVGL failed on Odroid Go..

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#19703] [Q&A] [answered]
Hi all,

I managed to install Tasmoto32, compiled for Odroid Go on the TasmoCompiler on an Adroid Go, set the "Autoconfig" to Odroid Go.

So far, so good.. Tasmota shows a grey screen, I can change the brightness with the slider.

Next thing I wanted to try out is to make the screen show something like described here, I installed the /autoexec.be

https://tasmota.github.io/docs/LVGL_in_10_minutes/#try-a-lvgl-demo-app 

Nothing I just get "lv undeclared" next to other error messages in the console..

    00:00:00.002 HDW: ESP32-D0WD (PSRAM)
    00:00:00.094 UFS: FlashFS mounted with 12580 kB free
    00:00:00.125 CFG: Loaded from File, Anzahl 25
    00:00:00.134 QPC: Count 1
    00:00:00.146 SPI: Hardware using GPIO18(CLK), GPIO23(MOSI) and GPIO19(MISO)
    00:00:00.148 I2C: Bus1 using GPIO04(SCL) and GPIO15(SDA)
    00:00:01.318 SCR: ufilesystem found
    00:00:01.334 TFS: File 'mcp23x.dat' not found
    00:00:01.453 BRY: Berry initialized, RAM used=4274 bytes
    00:00:01.526 TFS: File '.drvset003' not found
    00:00:01.538 Projekt tasmota - odroid-go-3 Version 13.1.0(TasmoCompiler-esp32odroid-go)-2_0_11(2023.10.08 10:44:36)
    00:00:01.967 BRY: failed to load '/autoexec.be' (syntax_error - /autoexec.be:2: 'lv' undeclared (first use in this function))
    00:00:02.672 WIF: verbinden mit AP1 dellekom Channel 6 BSSId <secret> in Modus 11n als odroid-go-3...
    00:00:03.333 WIF: verbunden
    00:00:03.605 HTP: Web-Server aktiv bei odroid-go-3 mit IP-Adresse 192.168.13.86
    12:33:15.659 RSL: INFO1 = {"Info1":{"Module":"ODROID_GO_I2C","Version":"13.1.0(TasmoCompiler-esp32odroid-go)","FallbackTopic":"cmnd/DVES_D42819_fb/","GroupTopic":"cmnd/tasmotas/"}}
    12:33:15.673 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"odroid-go-3","IPAddress":"192.168.13.86","IP6Global":"2003:a:<secret>","IP6Local":"fe80::b6e6:2dff:fed4:2819"}}
    12:33:15.685 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":9}}
    12:33:15.698 RSL: RESULT = {"POWER":"ON"}
    12:33:15.699 RSL: POWER = ON
    12:33:19.990 QPC: Reset
    12:33:19.997 RSL: STATE = {"Time":"2023-10-08T12:33:19","Uptime":"0T00:00:07","UptimeSec":7,"Heap":130,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":39},"POWER":"ON","Dimmer":65,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"dellekom","BSSId":"<secret>","Channel":6,"Mode":"11n","RSSI":58,"Signal":-71,"LinkCount":1,"Downtime":"0T00:00:02"}}
    12:33:20.075 RSL: SENSOR = {"Time":"2023-10-08T12:33:20","Switch5":"ON","Switch6":"ON","ANALOG":{"Joy1":0,"Joy2":0,"Range3":3722}}
   12:33:23.465 RSL: RESULT = {"POWER":"ON","Dimmer":68}

Any help welcome!

Andreas

**Comment by sfromis** [comment]
You may have done a build without selecting the LVGL feature? Maybe consider also including HASPmota, which makes many things easier for displays.

**Reply by hagenbuch** [reply, marked as answer]
I thought when I select "Odroid Go" in the Tasmocompiler, that alone switches it on? Tasmocompiler offers no "LVGL" checkbox, so I compiled it again, added

    #define USE_LVGL
    #define USE_DISPLAY
    #define USE_DISPLAY_LVGL_ONLY
    #define USE_XPT2046
    #define USE_UNIVERSAL_DISPLAY
      #undef USE_DISPLAY_MODES1TO5
      #undef USE_DISPLAY_LCD
      #undef USE_DISPLAY_SSD1306
      #undef USE_DISPLAY_MATRIX
      #undef USE_DISPLAY_SEVENSEG


Also from the back of my mind, I meant I have seen LVGL libraries being included in the comilation the first time.

Aber: HAT GEKLAPPT in dem Sinne daß ich einen blauen Hinterdrung mit rotes Kopfzeile "Tasmota" sehe mit Uhrzeit, darunter einen Button nach links, einen nach rechts und einen mit einem Haken.. mal sehen was das bedeutet. Danke! Müßte man vielleicht mal bei den Tasmocompiler-Leuten reingeben..

Ich hab leider immer noch 

    00:00:00.550 TFS: File 'mcp23x.dat' not found
    00:00:00.723 TFS: File '.drvset003' not found

aber jetzt

    00:00:01.084 LVG: LVGL initialized
    00:00:01.959 RSL: RESULT = {"DisplayDimmer":50}
    00:00:02.110 BRY: Successfully loaded 'autoexec.be'

**Comment by Jason2866** [comment]
Better to start with the pre built `Tasmota32-lvgl` firmware.

**Reply by hagenbuch** [reply]
I did now, thanks.. will have to RTFM :)

**Comment by Jason2866** [comment]
Is not an error. just an info. This files are used when using MCP230XX and when energy driver 03 is used
```
00:00:00.550 TFS: File 'mcp23x.dat' not found
00:00:00.723 TFS: File '.drvset003' not found
```

**Reply by hagenbuch** [reply]
Ah, thank you!
